jetbrains-core/it/software/aws/toolkits/jetbrains/services/cloudformation/annotations/ErrorAnnotatorTest.kt

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudformation.annotations
+
+import com.intellij.lang.annotation.Annotation
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiFile
+import com.nhaarman.mockitokotlin2.whenever
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito
+import java.io.File
+import java.io.IOException
+
+class ErrorAnnotatorTest {
```

Comment:
We never mock the JB APIs, their behavior is too undocumented and can change between versions.

`JavaCodeInsightTestFixtureRule` can be used to load any file (its not really Java tied, it just has a Java SDK configured in the project) and can be used to run annotators against loaded documents. Though I have never used it against an external annotator, but it should work.